### PR TITLE
Implement layered SandboxConfig with TOML tests and BDD coverage

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Add layered SandboxConfig support with TOML serialisation/deserialisation tests
- Extend BDD coverage with sandbox configuration scenarios and step definitions
- Update documentation to reflect SandboxConfig usage and defaults
- Mark SandboxConfig task as complete in the roadmap

### Core functionality
- Introduced comprehensive tests for SandboxConfig in src/config.rs, including:
  - TOML serialisation round-trips
  - All boolean combinations of `privileged` and `mount_dev_fuse`
  - Deserialisation from TOML into AppConfig and usage of defaults when the section is omitted
- Ensured default values are preserved (as per existing implementation):
  - `privileged: false`, `mount_dev_fuse: true`
- Clarified inline documentation for SandboxConfig defaults and behavior (no `validate()` method required since booleans always form valid states)

### Tests
- Added unit tests to test SandboxConfig TOML serialisation and round-trip
- Added parameterised tests for all four boolean combinations of SandboxConfig
- Extended existing AppConfig deserialisation tests to include sandbox defaults and omissions
- Updated tests/bdd_config.rs with new SandboxConfig step definitions

### Behavior-Driven Development (BDD)
- Added new scenarios to tests/features/configuration.feature:
  - Sandbox configuration with dev/fuse disabled
  - Sandbox configuration in minimal mode
  - Sandbox configuration in privileged mode with all options
- Implemented new step definitions in tests/bdd_config.rs:
  - Given a configuration file with dev/fuse mounting disabled
  - Given a configuration file in minimal mode
  - Given a configuration file with privileged mode and dev/fuse disabled
  - Then dev/fuse mounting is disabled
- Scenario bindings added for new sandbox scenarios to integrate with the existing configuration flow

### Documentation
- Added docs/execplans/1-3-3-sandbox-config.md as the SandboxConfig execplan document
- Updated docs/users-guide.md to reflect sandbox configuration, including:
  - Minimal mode defaults
  - Privileged mode trade-offs
  - /dev/fuse mounting guidance
- Updated docs/podbot-roadmap.md to mark SandboxConfig task as completed

## Validation plan
- Run:
  - make check-fmt
  - make lint
  - make test
- Expected: all commands pass with no warnings or errors
- Rationale: Tests cover TOML serialisation/deserialisation, defaulting behavior, and BDD scenarios for sandbox configuration

## How this satisfies the task
- Establishes SandboxConfig support as a layered configuration surface with clear defaults
- Provides comprehensive unit tests for TOML serialisation and combinations
- Adds BDD coverage to exercise sandbox-related configurations and edge cases
- Documents usage, trade-offs, and defaults, and reflects completion in the roadmap

🌿 Generated by [Terry](https://www.terragonlabs.com)

---

ℹ️ Tag @terragon-labs to ask questions and address PR feedback

📎 **Task**: https://www.terragonlabs.com/task/8d3d872c-5e2e-473d-bb96-0baadbfbb043

## Summary by Sourcery

Add comprehensive testing and documentation for SandboxConfig, including TOML behaviour, BDD scenarios, and roadmap updates.

Enhancements:
- Extend SandboxConfig coverage with TOML serialisation/deserialisation and default-handling tests.
- Add BDD step definitions and scenarios for sandbox configuration modes and fuse mounting behaviour.
- Document sandbox configuration modes, defaults, and trade-offs in the user guide and execplan, and mark the roadmap task as complete.

Documentation:
- Document SandboxConfig usage, defaults, and security/compatibility trade-offs in the user guide and a dedicated execplan.

Tests:
- Add unit tests for SandboxConfig TOML round-trips, all boolean flag combinations, and AppConfig deserialisation defaults.
- Introduce BDD scenarios and bindings exercising sandbox minimal, privileged, and dev/fuse-disabled configurations.

Chores:
- Update the roadmap to mark SandboxConfig configuration work as completed.
</body>

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[src/config.rs](https://github.com/leynos/podbot/pull/14/files/c7ef0413e4686fbca6a0a17ba0070b0f76693345#diff-cba64c21ab992eaad29fce147a08f4560a4769bc14682b8a96081a5fd02dbecd)

Comment on lines +672 to +683

~~~rust
    fn sandbox_config_uses_defaults_for_missing_fields() {
        let toml = r"
            [sandbox]
            privileged = true
        ";
        let config: AppConfig = toml::from_str(toml).expect("TOML parsing should succeed");
        assert!(config.sandbox.privileged, "Expected privileged to be true");
        assert!(
            config.sandbox.mount_dev_fuse,
            "Expected mount_dev_fuse to default to true when omitted"
        );
    }
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/75835/delta?repo-id=1423864&review-id=14&biomarker=Code+Duplication&filename=src%2Fconfig.rs&method=)
The module contains 3 functions with similar structure: tests.sandbox_config_deserialises_from_toml_section,tests.sandbox_config_uses_defaults_for_missing_fields,tests.sandbox_config_uses_defaults_when_section_omitted

File: src/config.rs [hunk 1]
```diff
@@ -585,4 +585,90 @@ mod tests {
...
+        let config: AppConfig = toml::from_str(toml_input).expect("TOML parsing should succeed");
+        assert_eq!(
+            config.sandbox.privileged, expected_privileged,
+            "{description}: privileged mismatch"
+        );
+        assert_eq!(
+            config.sandbox.mount_dev_fuse, expected_mount_dev_fuse,
+            "{description}: mount_dev_fuse mismatch"
+        );
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/config.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
